sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -375,11 +393,157 @@ cdef class DistanceMetric{{name_suffix}}:
                 D[i1, i2] = self.dist(&X[i1, 0], &Y[i2, 0], X.shape[1])
         return 0
 
-    cdef DTYPE_t _rdist_to_dist(self, {{DTYPE_t}} rdist) nogil except -1:
+    cdef DTYPE_t dist_csr(
+        self,
+        const {{INPUT_DTYPE_t}}[:] x1_data,
+        const SPARSE_INDEX_TYPE_t[:] x1_indices,
+        const {{INPUT_DTYPE_t}}[:] x2_data,
+        const SPARSE_INDEX_TYPE_t[:] x2_indices,
+        const SPARSE_INDEX_TYPE_t x1_start,
+        const SPARSE_INDEX_TYPE_t x1_end,
+        const SPARSE_INDEX_TYPE_t x2_start,
+        const SPARSE_INDEX_TYPE_t x2_end,
```

Comment:
It would be nice to have a description of theses somewhere in the docstrings, just once.
x_data and x_indices is the 2-d sparse array. So why to I need x_start and x_end?

Reply:
Does 731370a0fad51c50a7e95ba2a52da8c75721fd17 clarifies this?

Reply:
A comment was added that explains those parameters. Maybe we can do a little better, but I considered it as resolved and therefore merged.

Reply:
@ogrisel: did you mean the proposal of 731370a was insufficient, out of scope, or could be rephrased? :)